### ISSUE TITLE
Upgrade to Jetty 10.x (or Jetty 11.x)

### ISSUE THREAD

**Issue by joakime** [openhab/openhab-core#3315]
Jetty 9.x is now at End of Community Support.

You should be using Jetty 10.x (or newer) at this point.

* https://github.com/eclipse/jetty.project/issues/7958

**Comment by J-N-K** [comment]
We are currently upgrading Karaf to 4.4.3 with PaxWeb 8.0.15 (i.e. Jetty 9.4.50). There is some work on Karaf 4.5.x which will support PaxWeb 9 and Jetty 10 (https://lists.apache.org/thread/4lwbt9g0w707zomnq6cfx179n8w90d3h) but we'll have to check when we see a release there.

**Comment by andrewfg** [comment]
@J-N-K 4yi, this relates to..

- https://github.com/openhab/openhab-addons/pull/13570
- https://github.com/eclipse/jetty.project/issues/9168

**Comment by wborn** [comment]
The upgrade in Karaf is tracked by issue [KARAF-7544](https://issues.apache.org/jira/browse/KARAF-7544).

**Comment by jlaur** [comment]
I was checking release notes in relation to #3770 and saw this: https://issues.apache.org/jira/browse/CXF-8764. So for this library we are now at Jetty 10? So we are now bundling both Jetty 9 and 10 because of this dependency? Is it possible in the same way to decouple the dependency for PaxWeb, so that we can use Jetty 10 at the same time as PaxWeb still uses Jetty 9?

**Comment by J-N-K** [comment]
No, we currently only ship Jetty 9, as far as I can see we don't use the bundles that require jetty 10.

We could try if we can upgrade to PaxWeb 9 after Karaf 4.4 (because then we have SLF4J 2.0 available).

**Comment by joakime** [comment]
BTW, Jetty 12 was released.   (with support for EE8 at `java.servlet`, EE9 at `jakarta.servlet`, and EE10 at `jakarta.servlet` at the same time)

**Comment by lolodomo** [comment]
Any idea what will be required to change in bindings when moving to Jetty 10 ?

**Comment by J-N-K** [comment]
From my understanding changes are only required for websockets, see https://eclipse.dev/jetty/documentation/jetty-10/programming-guide/index.html#pg-migration

**Comment by lolodomo** [comment]
Ok, we could then search in code for all the classes that have to be renamed.

**Comment by andrewfg** [comment]
^ FWIW when I wrote the Hue API v2 binding, I had to use quite low level HTTP2 code from Jetty 9.x because 10.x was not available. I am guessing that if OH shifts to Jetty 10 that low level code will not be broken. However there may be some opportunity to optimise the binding with newer Jetty 10 higher level features. This observation is fairly hypothetical. But I would look at it in detail once we get the first snapshots with Jetty 11..

**Comment by jlaur** [comment]
Probably there will also be some things to be sorted out with dependencies, i.e. Karaf requiring Jetty 9?

**Comment by J-N-K** [comment]
Correct, well upgrade when Karaf upgrades, and I think this is scheduled for Karaf 4.5.0.

**Comment by joakime** [comment]
This issue has been open long enough that Jetty 10 and Jetty 11 are now at End of Community Support.

* https://github.com/jetty/jetty.project/issues/10485

Jetty 12 is the only supported version of Jetty at this point in time.
